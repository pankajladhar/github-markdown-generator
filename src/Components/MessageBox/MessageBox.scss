.messageBox {
  transition: transform .3s ease-out;
  position: fixed;
  top: -100px;
  left: 50%;
  padding: 15px;
  margin-top: 10px;
  border: 1px solid transparent;
  border-radius: 4px;
  z-index: 99;
  width: 550px;

  &.show {
    transform: translate(-50%, 100px);
  }

  &.hide {
    transform: translate(-50%, -100px);
  }

  .close-btn {
    position: absolute;
    right: 10px;
    top: 50%;
    transform: translateY(-50%);
    padding: 10px;
    cursor: pointer;
    font-size: 18px;
    color: inherit;
    opacity: .2;
    display: inline-block;
    line-height: 1;
    box-shadow: none;
    background: inherit;
    border: none;

    &:hover {
      opacity: 1;
    }
  }

  &.error {
    background: red;
    color: #a94442;
    background-color: #f2dede;
    border-color: #ebccd1;
  }

  &.success {
    color: #3c763d;
    background-color: #dff0d8;
    border-color: #d6e9c6;
  }
}
