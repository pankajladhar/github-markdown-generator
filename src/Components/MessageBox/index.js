import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Button from './../Button';
import './MessageBox.scss';

const propTypes = {
    title: PropTypes.string.isRequired,
    type: PropTypes.oneOf(['error', 'success']),
    showMessageBox: PropTypes.bool,
    autoClosingTimeInterval: PropTypes.number
}

const defaultProps = {
    showMessageBox: false,
    autoClosingTimeInterval: 2000
}

export default class MessageBox extends PureComponent {
    
    constructor(props){
        super(props)
    }
    
    componentDidMount(){
        setTimeout(()=> {
            this.props.closeBtnProps.handleClick();
        }, this.props.autoClosingTimeInterval);
    }

    getMessageBoxClasses(){
        let isShowing = this.props.showMessageBox == true ? "show" : "hide";
        return `alert ${this.props.type} ${isShowing}`;
    }

    render() {
        return (
            <div className={this.getMessageBoxClasses()}>
                {
                    this.props.showMessageBox &&
                        <div>
                            {this.props.title}
                            <Button className="close-btn"
                                title={this.props.closeBtnProps.title}
                                value={this.props.closeBtnProps.value}
                                handleClick={this.props.closeBtnProps.handleClick}   
                            />
                        </div>
                }
            </div>
        );
    }
}

MessageBox.propTypes = propTypes;
MessageBox.defaultProps = defaultProps;
