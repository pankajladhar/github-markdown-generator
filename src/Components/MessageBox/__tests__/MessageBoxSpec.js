import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import MessageBox from './../index';

describe('MessageBox', ()=>{

    let messageBoxProps = {};

    beforeEach(()=>{
        messageBoxProps = {
            title: "This is message Box",
            type: "success",
            showMessageBox: true,
            autoClosingTimeInterval:1000,
            closeBtnProps: {
                title: "close",
                value: "x",
                handleClick: () =>{}
            }
        }
    })

    it('should not render when value of showMessageBox is not provided or false ', ()=>{
        messageBoxProps.showMessageBox = false;
        let tree = renderer.create(
            <MessageBox {...messageBoxProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when value of showMessageBox is true', ()=>{
        let tree = renderer.create(
            <MessageBox {...messageBoxProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.props.className.indexOf("success")).toBeGreaterThan(0)
    });

    it('should have class error when type props is error', ()=>{
        messageBoxProps.type = "error";
        let tree = renderer.create(
            <MessageBox {...messageBoxProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.props.className.indexOf("error")).toBeGreaterThan(0)
    });

    it("should gets close automatically", ()=>{
        jest.useFakeTimers();
        let mockHandleClick = jest.fn();

        messageBoxProps.closeBtnProps.handleClick = mockHandleClick;
        
        expect(mockHandleClick).not.toBeCalled();
        
        const spy = jest.spyOn(MessageBox.prototype, 'componentDidMount');
        const wrapper = mount(<MessageBox {...messageBoxProps} />);
        wrapper.instance().componentDidMount();
        expect(spy).toHaveBeenCalled();
        expect(setTimeout.mock.calls[0][1]).toBe(1000);
        
        jest.runAllTimers();
        expect(mockHandleClick).toBeCalled();
    })
});
