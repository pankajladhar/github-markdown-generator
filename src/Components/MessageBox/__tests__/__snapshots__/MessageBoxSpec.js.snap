// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox renders correctly when value of showMessageBox is true 1`] = `
<div
  className="messageBox success show"
>
  <div>
    This is message Box
    <button
      className="button close-btn"
      disabled={false}
      name={undefined}
      onClick={[Function]}
      title="close"
    >
      x
    </button>
  </div>
</div>
`;

exports[`MessageBox should have class error when type props is error 1`] = `
<div
  className="messageBox error show"
>
  <div>
    This is message Box
    <button
      className="button close-btn"
      disabled={false}
      name={undefined}
      onClick={[Function]}
      title="close"
    >
      x
    </button>
  </div>
</div>
`;

exports[`MessageBox should not render when value of showMessageBox is not provided or false  1`] = `
<div
  className="messageBox success hide"
/>
`;
